packages/block-library/src/image/edit.js

```diff
@@ -333,8 +334,8 @@ export function ImageEdit( {
 		return (
 			<>
 				{ controls }
-				<li className="list-image">
-					<figure { ...blockProps }>
+				<li { ...blockProps }>
```

Comment:
In my opinion, a block should not be aware of the context it's in, so we shouldn't be adding list item elements at this level. This needs to be the responsibility of the parent block. If we want to continue with having a list in the gallery, we should fix inner block parsing to ignore wrapper elements around blocks. Alternatively, could we avoid lists entirely here?

Reply:
> Not sure if it's been discussed before, but is there any way we can avoid lists at all in the gallery block and only have nested figure elements?

Thanks for the feedback @ellatrix. I initially structured it as a collection of `figures`, but changed to a `<li>` based on this [accessibility related feedback](https://github.com/WordPress/gutenberg/pull/24039#issuecomment-663093575). This does make sense to me from a semantic structure point of view to me, but I also don't like the need to make the image block aware of how it should be wrapped, so very open to ideas/thoughts on the best way forward here.

The approach that @ZebulanStanphill [suggested here](https://github.com/WordPress/gutenberg/pull/24039#discussion_r461323119) worked well, apart from the issue of the block validation failing ... how much work would be involved in reimplementing this along with a fix for the validation issue.

On a similar not about the block being aware of context @ellatrix, we were pushing any changes to the Image linkDestination, linkTarget and sizeSlug down to the children via modifying the innerBlocks, but there was an issue with this approach in not knowing if a user wanted any changes made to individual Image blocks overwritten or not ... to get around it this PR has a prompt in the settings sidebar to ask if user wants the change applied to all child images when the settings are changed. This is a bit of any antipattern as it means settings panel changes are not immediately applied, so it was suggested that we let the child Image handle these changes via block context. 

This [PR](https://github.com/WordPress/gutenberg/pull/27726) makes this change so that the Image block applies certain attribute changes based on context changes, and also sets a flag if these attributes are changed at the Image level so that changes to the parent don't overwrite them ... possible use case is:

- User adds Gallery with 3 Images
- User edits one Image to add a custom URL link
- User then sets Gallery link default to attachment and expects that 2 Images will get this change, but the one with custom url will remain as is

Do you have any thoughts on the Image block being context aware in this instance?

Reply:
That sounds more like a parent component controlling the attributes child blocks, rather than context? Or does that not work?

Regarding the list items, I think the PR that @ZebulanStanphill was great, we just need to fix block parsing to ignore markup surrounding inner blocks.

Reply:
Scratch that, [we should just avoid the list items](https://github.com/WordPress/gutenberg/pull/25940#issuecomment-746334889).